src/main.rs

```diff
@@ -813,6 +853,157 @@ async fn handle_request(
+                slog::warn!(
+                    logger,
+                    "Could not load cert `{}`: {}",
+                    cert_path.display(),
+                    e
+                );
+                continue;
+            }
+            match cert_path.extension() {
+                Some(v) if v == "pem" => {
+                    slog::info!(
+                        logger,
+                        "adding PEM cert `{}` to root certificates",
+                        cert_path.display()
+                    );
+                    let mut pem = Cursor::new(buf);
+                    let certs = match rustls_pemfile::certs(&mut pem) {
+                        Ok(v) => v,
+                        Err(e) => {
+                            slog::warn!(
+                                logger,
+                                "No valid certificate was found `{}`: {}",
+                                cert_path.display(),
+                                e
+                            );
+                            continue;
+                        }
+                    };
+                    for c in certs {
+                        if let Err(e) = root_cert_store.add(&rustls::Certificate(c)) {
+                            slog::warn!(
+                                logger,
+                                "Could not add part of cert `{}`: {}",
+                                cert_path.display(),
+                                e
+                            );
+                        }
+                    }
+                }
+                Some(v) if v == "der" => {
+                    slog::info!(
+                        logger,
+                        "adding DER cert `{}` to root certificates",
+                        cert_path.display()
+                    );
+                    if let Err(e) = root_cert_store.add(&Certificate(buf)) {
+                        slog::warn!(
+                            logger,
+                            "Could not add cert `{}`: {}",
+                            cert_path.display(),
+                            e
+                        );
+                    }
+                }
+                _ => slog::warn!(
+                    logger,
+                    "Could not load cert `{}`: unknown extension",
+                    cert_path.display()
+                ),
+            }
+        }
+
+        use rustls::OwnedTrustAnchor;
+        let trust_anchors = webpki_roots::TLS_SERVER_ROOTS.0.iter().map(|trust_anchor| {
+            OwnedTrustAnchor::from_subject_spki_name_constraints(
+                trust_anchor.subject,
+                trust_anchor.spki,
+                trust_anchor.name_constraints,
+            )
+        });
+        root_cert_store.add_server_trust_anchors(trust_anchors);
+
+        builder
+            .with_root_certificates(root_cert_store)
+            .with_no_client_auth()
+    } else {
+        use rustls::client::HandshakeSignatureValid;
```

Comment:
Would it make sense to simplify this `if/else` block by pulling out some stuff to separate functions?

}
```